pkg/controller/memberstatus/memberstatus_controller.go

```diff
@@ -226,24 +280,32 @@ func (r *ReconcileMemberStatus) memberOperatorHandleStatus(reqLogger logr.Logger
 		memberStatus.Status.MemberOperator = operatorStatus
 		return err
 	}
+	operatorStatus.Deployment.Name = memberDeployment.Name
+
+	// check replicas
+	if memberDeployment.Status.UnavailableReplicas > 0 {
```

Comment:
Have you checked that `UnavailableReplicas > 0` only if there is a problem with some pods?
Let's make sure `UnavailableReplicas` is 0 in case of normal a rolling update when a new pod is still starting but not fully ready yet.

Reply:
Looks like during a normal rolling update `UnavailableReplicas` can be >0. Not sure how we want to handle this situation. Maybe we can go back to only checking the conditions and create a separate issue to come back to this to beef up how we detect problems with deployment rollouts?

Reply:
> Maybe we can go back to only checking the conditions and create a separate issue to come back to this to beef up how we detect problems with deployment rollouts?

+1. Let's do it.

Reply:
Done